src/account/UpgradeableModularAccount.sol

```diff
@@ -573,6 +616,41 @@ contract UpgradeableModularAccount is
     // solhint-disable-next-line no-empty-blocks
     function _authorizeUpgrade(address newImplementation) internal override {}
 
+    function _checkIfValidationApplies(bytes4 selector, FunctionReference validationFunction, bool isDefault)
+        internal
+        view
+    {
+        AccountStorage storage _storage = getAccountStorage();
+
+        // Check that the provided validation function is applicable to the selector
+        if (isDefault) {
+            if (
+                !_defaultValidationAllowed(selector)
+                    || !_storage.defaultValidations.contains(toSetValue(validationFunction))
+            ) {
+                revert UserOpValidationFunctionMissing(selector);
+            }
+        } else {
+            // Not default validation, but per-selector
```

Comment:
Minor, during the installation, it seems like we also install the default validation 
```
for (uint256 i = 0; i < selectors.length; ++i) {
            bytes4 selector = selectors[i];
            if (!_storage.selectorData[selector].validations.add(toSetValue(validationFunction))) {
                revert ValidationAlreadySet(selector, plugin, functionId);
            }
        }
```
should we check `getAccountStorage().selectorData[selector].validations.contains()` for both default and non-default?

Reply:
That path in the installation is used if the user wishes to install a validation both as a default validation and for some specific selectors. If the user manually adds selectors that are already usable from default validation (e.g. `execute`), then the cost savings of default validation go away. Similarly, checking the per-selector storage for the default validation increases costs due to the storage reads without providing a new form of checking.

Reply:
I see, that makes sense!